code/modules/detective_work/swabs.dm

```diff
@@ -0,0 +1,171 @@
+/obj/item/clothing/clean_blood(radiation_clean = FALSE)
+	. = ..()
+	gunshot_residue = null
+
+/obj/item/forensics/swab
+	name = "\improper sample collection kit"
+	desc = "Sterile cotton swab and test tube for collecting samples."
+	icon_state = "swab"
+	var/dispenser = FALSE
+	var/gsr = FALSE
+	var/list/dna
+	var/used
+	var/inuse = FALSE
+
+/obj/item/forensics/swab/proc/is_used()
+	return used
+
+/obj/item/forensics/swab/attack(mob/living/M, mob/user)
+	if(!ishuman(M))
+		return ..()
+	if(is_used())
+		to_chat(user, "<span class='warning'>You are already taking a sample.</span>")
+		return
+
+	var/mob/living/carbon/human/H = M
+	var/sample_type
+	inuse = TRUE
+	to_chat(user, "<span class='notice'>You start collecting samples.</span>")
+	if(do_after(user, 2 SECONDS, target = user))
+		if(H.wear_mask)
+			to_chat(user, "<span class='warning'>[H] is wearing a mask.</span>")
+			inuse = FALSE
+			return
```

Comment:
Probably better to check both before and after the do-after. I think you might be able to add a custom callback that checks this dynamically, though, so people can cancel it by putting on a mask